pkg/operator/ceph/csi/controller.go

```diff
@@ -140,16 +150,36 @@ func (r *ReconcileCSI) reconcile(request reconcile.Request) (reconcile.Result, e
 
 		return reconcile.Result{}, nil
 	} else {
-		for _, cluster := range cephClusters.Items {
+		for i, cluster := range cephClusters.Items {
 			if !cluster.DeletionTimestamp.IsZero() {
-				logger.Debug("ceph cluster is being deleting, no need to reconcile the csi driver")
+				logger.Debugf("ceph cluster %q is being deleting, no need to reconcile the csi driver", cluster.Name)
 				return reconcile.Result{}, nil
 			}
 
 			if !cluster.Spec.External.Enable && cluster.Spec.CleanupPolicy.HasDataDirCleanPolicy() {
-				logger.Debug("ceph cluster has cleanup policy, the cluster will soon go away, no need to reconcile the csi driver")
+				logger.Debugf("ceph cluster %q has cleanup policy, the cluster will soon go away, no need to reconcile the csi driver", cluster.Name)
 				return reconcile.Result{}, nil
 			}
+
+			// Do we have a multus cluster?
+			// If so deploy the plugin holder with the fsid attached
+			if cluster.Spec.Network.IsMultus() {
+				// Load cluster info for later use in updating the ceph-csi configmap
+				clusterInfo, _, _, err := opcontroller.LoadClusterInfo(r.context, r.opManagerContext, cluster.Namespace)
```

Comment:
Why do we need to load the cluster info here and cache it? Why not just load the info later when it's needed for updating the csi info? Or maybe we just cache the cephClusters in the slice and load the cluster info later when needed?

Reply:
We load clusterInfo to get the clusterFSID as well as the mon endpoints. Essentially later when building the spec code, see `spec.go`, the for loop that builds the holder daemonset. I was thinking it was nicer to have everything done here instead of calling LoadClusterInfo later. At the end, it's more or less the same...